server/src/main/java/org/elasticsearch/common/geo/GeoPoint.java

```diff
@@ -75,14 +77,20 @@ public GeoPoint resetLon(double lon) {
     }
 
     public GeoPoint resetFromString(String value) {
-        int comma = value.indexOf(',');
-        if (comma != -1) {
-            lat = Double.parseDouble(value.substring(0, comma).trim());
-            lon = Double.parseDouble(value.substring(comma + 1).trim());
-        } else {
-            resetFromGeoHash(value);
+        return resetFromString(value, false);
+    }
+
+    public GeoPoint resetFromString(String value, final boolean ignoreZValue) {
+        if (value.contains(",")) {
+            String[] vals = value.split(",");
```

Comment:
please fail if vals.length > 3

Reply:
Wouldn't it better to be more lenient and accept the first 3 (or 2) dimensions, rather than completely fail? Adding failure now will break compatibility with previous behavior.